addons/binding/org.openhab.binding.ihc/src/main/java/org/openhab/binding/ihc/internal/converters/ConverterFactory.java

```diff
@@ -0,0 +1,143 @@
+/**
+ * Copyright (c) 2010-2018 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.ihc.internal.converters;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.eclipse.smarthome.core.library.types.DateTimeType;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.OpenClosedType;
+import org.eclipse.smarthome.core.library.types.PercentType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.library.types.UpDownType;
+import org.eclipse.smarthome.core.types.Type;
+import org.openhab.binding.ihc.internal.ws.resourcevalues.WSBooleanValue;
+import org.openhab.binding.ihc.internal.ws.resourcevalues.WSDateValue;
+import org.openhab.binding.ihc.internal.ws.resourcevalues.WSEnumValue;
+import org.openhab.binding.ihc.internal.ws.resourcevalues.WSFloatingPointValue;
+import org.openhab.binding.ihc.internal.ws.resourcevalues.WSIntegerValue;
+import org.openhab.binding.ihc.internal.ws.resourcevalues.WSResourceValue;
+import org.openhab.binding.ihc.internal.ws.resourcevalues.WSTimeValue;
+import org.openhab.binding.ihc.internal.ws.resourcevalues.WSTimerValue;
+import org.openhab.binding.ihc.internal.ws.resourcevalues.WSWeekdayValue;
+
+/**
+ * IHC / ELKO <-> openHAB data type converter factory.
+ *
+ * @author Pauli Anttila - Initial contribution
+ */
+public enum ConverterFactory {
+    INSTANCE;
```

Comment:
This is a really 'interesting' use of enum. Beter make it a class and make INSTANCE a static final field.

Reply:
E.g. https://dzone.com/articles/java-singletons-using-enum claims that enum is recommended as the best method of making singletons in Java.

Reply:
Ah ok. Seems I need to do some more studying :wink: